perun-core/src/main/java/cz/metacentrum/perun/core/impl/AuditMessagesManagerImpl.java

```diff
@@ -259,15 +253,21 @@ public void createAuditerConsumer(PerunSession perunSession, String consumerName
 
 	@Override
 	public Map<String, Integer> getAllAuditerConsumers(PerunSession sess) throws InternalErrorException {
-		Map<String, Integer> auditerConsumers;
-
 		try {
-			auditerConsumers = jdbc.query("select name, last_processed_id from auditer_consumers", AUDITER_CONSUMER_EXTRACTOR);
+			return jdbc.query("select name, last_processed_id from auditer_consumers", AUDITER_CONSUMER_EXTRACTOR);
 		} catch (RuntimeException ex) {
 			throw new InternalErrorException(ex);
 		}
+	}
 
-		return auditerConsumers;
+	@Override
+	public boolean checkAuditerConsumerExists(PerunSession session, String consumerName) throws InternalErrorException {
+		if (consumerName == null) throw new InternalErrorException("Auditer consumer doesn't exist.");
+		try {
+			return jdbc.queryForInt("select count(*) from auditer_consumers where name=?", consumerName) == 1;
```

Comment:
I am afraid there could be more than 1 consumer with such name. You should trow an exception if number of consumers is bigger than 1.

Reply:
There is unique constraint on the column, so we don't have to check on duplicate consumers in the code.

Reply:
You are right.